lib/fs.js

```diff
@@ -1338,11 +1345,11 @@ fs.appendFile = function(path, data, options, callback_) {
     throwOptionsError(options);
   }
 
-  if (!options.flag)
-    options = util._extend({ flag: 'a' }, options);
+  // Don't make changes directly on options object
+  options = copyObject(options);
```

Comment:
@yorkie Because of the performance https://github.com/nodejs/node/pull/7831#issuecomment-236356269